app/settings/user-settings/user-settings.controller.js

```diff
@@ -50,6 +50,7 @@ function (
     }
 
     function saveKey() {
+        $scope.api.$dirty = true;
         if ($scope.apiKey) {
             UserSettingsEndpoint.saveCache({id: $scope.hxlApiId,  user_id: $rootScope.currentUser.userId,  config_key: 'hdx_api_key', config_value: $scope.apiKey}).$promise.then((response) => {
                 $scope.hxlApiKey = '*** *** *** *** *** *** *** ' + $scope.apiKey.slice($scope.apiKey.length - 4);
```

Comment:
ah yea this is super sketchy, we should have done it in the formatter for sure.

Reply:
adding: obviously not a blocker to this, and I'm 90% sure it was a conscious decision at the time since we didn't have the formatter ready. @willdoran can we add to v1?

Reply:
Yep I'll make a ticket @rowasc